x-pack/plugins/fleet/server/services/epm/elasticsearch/template/install.ts

```diff
@@ -223,41 +221,60 @@ function putComponentTemplate(
   };
 }
 
-const mappingsSuffix = '@mappings';
-const settingsSuffix = '@settings';
-const userSettingsSuffix = '@custom';
 type TemplateBaseName = string;
-type UserSettingsTemplateName = `${TemplateBaseName}${typeof userSettingsSuffix}`;
+type UserSettingsTemplateName = `${TemplateBaseName}${typeof USER_SETTINGS_TEMPLATE_SUFFIX}`;
 
 const isUserSettingsTemplate = (name: string): name is UserSettingsTemplateName =>
-  name.endsWith(userSettingsSuffix);
+  name.endsWith(USER_SETTINGS_TEMPLATE_SUFFIX);
 
 function buildComponentTemplates(params: {
+  mappings: IndexTemplateMappings;
   templateName: string;
   registryElasticsearch: RegistryElasticsearch | undefined;
   packageName: string;
   defaultSettings: IndexTemplate['template']['settings'];
 }) {
```

Comment:
Overall, I'm finding the naming of all the different settings / mappings here hard to understand where they come from. Could we spend some time trying to improve the namings of the variables and types to make this more clear?